### PR TITLE
Change from idealOutput to evalFunction arg; report expected

### PR DESCRIPTION
**Pull request by benjreinhart**
Screenshot 2023-07-29 at 3 41 15 PM

### CODE REVIEW

**Comment by benjreinhart** [comment]
This also sets up next commit for runner

File: packages/axeval/example/completion.ts [hunk 1]
```diff
@@ -7,40 +7,32 @@ const dataset: CompletionEvalCase[] = [
   {
     description: 'Football world cup completion',
     prompt: 'Who won the 1998 football world cup? Respond concisly',
-    idealOutput: 'France',
-    evalFunctions: [new Match(), new Includes()],
+    evalFunctions: [new Match('France'), new Includes('France')],
```
Comment (nichochar) [inline packages/axeval/example/completion.ts:10]: I really like this better 👍

File: packages/axeval/src/evalFunction.ts [hunk 2]
```diff
@@ -2,36 +2,27 @@ import { OpenAIChat, SUPPORTED_OPENAI_CHAT_MODELS } from './model';
...
-  }
-
-  abstract run(response: string, idealOutput: string): Promise<number>;
+  options?: Record<string, any>;
+  expected: string;
+  run(response: string): Promise<number>;
 }
 
-export class IsValidJson extends BaseEvalFunction {
+export class IsValidJson implements EvalFunction {
```
Comment (nichochar) [inline packages/axeval/src/evalFunction.ts:12]: I guess the base class wasn't useful. Thanks for simplifying

File: packages/axeval/src/evalFunction.ts [hunk 3]
```diff
@@ -2,36 +2,27 @@ import { OpenAIChat, SUPPORTED_OPENAI_CHAT_MODELS } from './model';
...
-  constructor() {
-    super('Check if response is valid JSON');
+  description = 'Check if response is valid JSON';
+
+  get expected() {
+    return 'Response to be valid JSON';
   }
 
-  run(response: string): Promise<number> {
+  async run(response: string) {
```
Comment (nichochar) [inline packages/axeval/src/evalFunction.ts:20]: Aaaa I was wondering why my TypeScript editor was annoying me into coding up Promise resolves

**Review by nichochar — COMMENTED** [review]
awesome

**Pull request merged**